component/file_cache/file_cache.go

```diff
@@ -920,6 +1022,24 @@ func (fc *FileCache) ReadInBuffer(options internal.ReadInBufferOptions) (int, er
 	// The file should already be in the cache since CreateFile/OpenFile was called before and a shared lock was acquired.
 	// log.Debug("FileCache::ReadInBuffer : Reading %v bytes from %s", len(options.Data), options.Handle.Path)
 
+	options.Handle.Lock()
+	var err error
+	if dwnldIntfce, found := options.Handle.GetValue("isDownloadNeeded"); found {
```

Comment:
I think these other checks for Download could be simplified since I believe this logic is already in DownloadFile.